test/unit/math/rev/functor/algebra_solver_newton_test.cpp

```diff
@@ -0,0 +1,237 @@
+#include <stan/math/rev/core.hpp>
+#include <stan/math/rev/functor/algebra_solver_newton.hpp>
+#include <test/unit/math/rev/fun/util.hpp>
+#include <test/unit/math/rev/functor/util_algebra_solver.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+
+//////////////////////////////////////////////////////////////////////////
+// Tests for newton solver.
+
+TEST_F(algebra_solver_simple_eq_test, newton_dbl) {
+  bool is_newton = true;
+  Eigen::VectorXd theta = simple_eq_test(simple_eq_functor(), y_dbl, is_newton);
+}
+
+TEST_F(algebra_solver_simple_eq_test, newton_tuned_dbl) {
+  bool is_newton = true;
+  Eigen::VectorXd theta = simple_eq_test(simple_eq_functor(), y_dbl, is_newton,
+                                         true, scale_step, xtol, ftol, maxfev);
+}
+
+TEST_F(algebra_solver_simple_eq_nopara_test, newton_dbl) {
+  using stan::math::algebra_solver_newton;
+  Eigen::VectorXd theta = algebra_solver_newton(simple_eq_functor_nopara(), x,
+                                                y_dummy, dat, dummy_dat_int);
+  EXPECT_EQ(20, theta(0));
+  EXPECT_EQ(2, theta(1));
+}
+
+TEST_F(algebra_solver_non_linear_eq_test, newton_dbl) {
+  bool is_newton = true;
+  Eigen::VectorXd theta
+      = non_linear_eq_test(non_linear_eq_functor(), y_dbl, is_newton);
+  EXPECT_FLOAT_EQ(-y_dbl(0), theta(0));
+  EXPECT_FLOAT_EQ(-y_dbl(1), theta(1));
+  EXPECT_FLOAT_EQ(y_dbl(2), theta(2));
+}
+
+TEST_F(error_message_test, newton_dbl) {
+  bool is_newton = true;
+  error_conditions_test(non_linear_eq_functor(), y_3, is_newton);
+}
```

Comment:
So this is the one that's failing in `kinsol_data` on line 75 when we assign the results of `f_` to the Eigen map of the NVector.

```
    f_eval_map = apply(
        [&](const auto&... args) {
          return explicit_system->f_(x_eigen, explicit_system->msgs_, args...);
        },
        explicit_system->args_tuple_);
```

Trying to sort out why that's happening now